internal/tasks/templateversion_populate.go

```diff
@@ -31,14 +31,14 @@ func templateVersionPopulate(ctx context.Context, resourceRef *ResourceReference
 }
 
 type TemplateVersionPopulateLogic struct {
-	callbackManager CallbackManager
-	log             logrus.FieldLogger
-	store           store.Store
-	k8sClient       k8sclient.K8SClient
-	resourceRef     ResourceReference
-	templateVersion *api.TemplateVersion
-	fleet           *api.Fleet
-	frozenConfig    []api.TemplateVersionStatus_Config_Item
+	callbackManager    CallbackManager
+	log                logrus.FieldLogger
+	store              store.Store
+	k8sClient          k8sclient.K8SClient
+	resourceRef        ResourceReference
+	templateVersion    *api.TemplateVersion
+	fleet              *api.Fleet
+	frozenConfigSource []api.DeviceConfigSourceSpec
```

Comment:
Don't we want to freeze the hooks too ?

Reply:
Yeah I am not sure I did think about it a bit, what are the criteria for freezing?

Reply:
According to my understanding if some configuration is applied to a fleet - and after a month a device joins the fleet, we would like to the same configuration that was applied previously to other devices in the fleet should also be applied to the new device.

Reply:
good conversation for monday thanks for bringing it up

Reply:
going to need t be a followup